### ISSUE TITLE
Grails 3.2.0.RC2 two datasources second datasource No Session found for current thread

### ISSUE THREAD

**Issue by jknudsen** [grails/grails-data-mapping#795]
When I added a second data sources and when in development the second datasource gives me the error: **No Session found for current thread.**  When I run a integration test the second datasource works?  Both the test and development environments look to be configured the same.  Here is the application.yml:

---
```
hibernate:
    cache:
        queries: false
        use_second_level_cache: true
        use_query_cache: false
        region.factory_class: org.hibernate.cache.ehcache.SingletonEhCacheRegionFactory
    naming_strategy: 'org.hibernate.cfg.DefaultNamingStrategy'

dataSources:
    dataSource:
        pooled: true
        jmxExport: true
        driverClassName: org.h2.Driver
        username: sa
        password:
    secondDataSource:
        driverClassName: net.sourceforge.jtds.jdbc.Driver
        username: user
        dbCreate: none
environments:
    development:
        dataSources:
            dataSource:
                dbCreate: create-drop
                url: jdbc:h2:mem:devDb;MVCC=TRUE;LOCK_TIMEOUT=10000;DB_CLOSE_ON_EXIT=FALSE
            secondDataSource:
                password: 'xxxxx'
                url: jdbc:jtds:sqlserver://SERVER/DBName
    test:
        dataSources:
            dataSource:
                dbCreate: create-drop
                url: jdbc:h2:mem:devDb;MVCC=TRUE;LOCK_TIMEOUT=10000;DB_CLOSE_ON_EXIT=FALSE
            secondDataSource:
                password: 'xxxxx'
                url: jdbc:jtds:sqlserver://SERVER/DBName

```

**Comment by jknudsen** [comment]
Here is the Line of Code that is failing:

`List organizationList = Organization.findAll()`

**Comment by jknudsen** [comment]
I ended up making this one Datasource instead of two.  The Domain object then starting getting an error message about the View saying "view or function is not updatable because the modification affects multiple base tables".  Which could have been causing the problem.  The error message was not as good as when I used one datasource and why did the Functional Test work!  I am going to try again just using the Sql object to access that data in the second data source instead of a Domain object and I will report back on how that works.
